libraries/AP_GPS/AP_GPS.h

```diff
@@ -327,6 +336,8 @@ class AP_GPS
         return vertical_accuracy(primary_instance, vacc);
     }
 
+    CovarianceType position_covariance(const uint8_t instance, Matrix3d& cov) const WARN_IF_UNUSED;
```

Comment:
does it really need to be double precision? More stack space, and slow on F427
I know the final IDL msg is double, but I think we should not waste the stack and processor more than we have to

Reply:
Nope, it's only on the assignment. 

If the GPS receiver supported covariance as a double, it would be different, but that's not what the interface exposes today. 
Solved in fdd56d5c69477e1c02886d96f36deb4282276c1b

Reply:
what I meant was to change the AP_GPS API position_covariance() to take a Matrix3f reference, which will be smaller, faster and less stack. It's not a big deal, just trying to not move to double where we don't need to, and I can't really see that the extra precision for these covariances is of any benefit

Reply:
Done.